econml/utilities.py

```diff
@@ -582,6 +583,25 @@ def get_input_columns(X, prefix="X"):
     return [f"{prefix}{i}" for i in range(len_X)]
 
 
+def get_feature_names_or_default(featurizer, feature_names):
+    if hasattr(featurizer, 'get_feature_names'):
+        # Get number of arguments, some sklearn featurizer don't accept feature_names
+        arg_no = len(inspect.getfullargspec(featurizer.get_feature_names).args)
+        if arg_no == 1:
+            return featurizer.get_feature_names()
+        elif arg_no == 2:
+            return featurizer.get_feature_names(feature_names)
+    # Featurizer doesn't have 'get_feature_names' or has atypical 'get_feature_names'
+    try:
+        # Get feature names using featurizer
+        dummy_X = np.empty((1, len(feature_names)))
+        return get_input_columns(featurizer.transform(dummy_X))
```

Comment:
Minor suggestion, instead of using the default prefix X, might be better to use a prefix like "TransX" to make it clear it's the X after transformation. It's possible that after transformation the dimension of X might change, use the same prefix might be a bit confused.

Reply:
Good point! I think i will use `feat(X)` as a prefix for people to understand that it's the featurizer applied to `X` and avoid naming confusion.